src/main/java/org/broadinstitute/hellbender/utils/pairhmm/DragstrReadSTRAnalizer.java

```diff
@@ -0,0 +1,182 @@
+package org.broadinstitute.hellbender.utils.pairhmm;
+
+import org.broadinstitute.hellbender.utils.MathUtils;
+
+import java.util.Arrays;
+
+public class DragstrReadSTRAnalizer {
+    private final int[][] repeatsByPeriodAndPosition;
+    private final int[] periodWithMostRepeats;
+    private final int maxPeriod;
+    private int seqLength;
+
+    DragstrReadSTRAnalizer(final int maxSequenceLength, final int maxPeriod) {
+        repeatsByPeriodAndPosition = new int[maxPeriod][maxSequenceLength];
+        this.maxPeriod = maxPeriod;
+        this.periodWithMostRepeats = new int[maxSequenceLength];
+    }
+
+    public int numberOfRepeats(final int position, final int period) {
+        if (period <= 0 || period > maxPeriod) {
+            return 0;
+        } else if (position < 0 || position >= seqLength) {
+            throw new IllegalArgumentException("cannot query outside requested boundaries");
+        } else {
+            return repeatsByPeriodAndPosition[period - 1][position];
+        }
+    }
+
+    public int mostRepeatedPeriod(final int position) {
+        if (position >= 0 && position < seqLength) {
+            return periodWithMostRepeats[position];
+        } else {
+            throw new IllegalArgumentException("cannot query outside requested boundaries");
+        }
+    }
+
+    public int numberOfMostRepeats(final int position) {
+        if (position >= 0 && position < seqLength) {
+            return repeatsByPeriodAndPosition[periodWithMostRepeats[position] - 1][position];
+        } else {
+            throw new IllegalArgumentException("cannot query outside requested boundaries");
+        }
+    }
+
+    public void load(final byte[] sequence) {
```

Comment:
rename this "sequence" to "bases" to make it clearer this is intended to be run on read bases